hayeon/programmers-1/옹알이-2.js

```diff
@@ -0,0 +1,32 @@
+function solution(babbling) {
+  let result = 0;
+
+  const doubleBabbling = ["ayaaya", "yeye", "woowoo", "mama"];
+
+  while (babbling.length) {
+    let sound = babbling.pop();
+
+    if (doubleBabbling.some((el) => sound.includes(el))) continue;
+
+    sound = sound
+      .replaceAll("aya", "1")
+      .replaceAll("ye", "2")
+      .replaceAll("woo", "3")
+      .replaceAll("ma", "4");
+
+    sound = sound.replace(/[1234]/g, "");
+
+    if (sound.length === 0) result++;
+  }
+
+  return result;
+}
```

Comment:
```suggestion
function solution(babbling) {
  const singleRegex = /(aya|ye|woo|ma)/g;
  let result = 0;

  const doubleBabbling = ["ayaaya", "yeye", "woowoo", "mama"];

  while (babbling.length) {
    let sound = babbling.pop();

    if (doubleBabbling.some((el) => sound.includes(el))) continue;

    sound = sound.replace(singleRegex, '');

    if (!sound.length) result++;
  }

  return result;
}
```
- 옹알이2 풀이 잘봤습니다 🙇‍♂️
- sound.replaceAll 메서드 체이닝으로 반복하는 것 보다 아래 정규표현식 이용했듯이 정규표현식을 이용하시면 훨씬 간단한 풀이가 가능합니다.
- 아래는 제 정규표현식을 적극적으로 활용한 풀이입니다.
```js
function solution(babbling) {
  const singleRegex = /(aya|ye|woo|ma)/g;
  const doubleRegex = /(ayaaya|yeye|woowoo|mama)/g;

  return babbling.reduce((acc, cur) => {
    const replacedDoubleBabbling = cur.replace(doubleRegex, "X");
    const replacedSingleBabbling = replacedDoubleBabbling.replace(
      singleRegex,
      ""
    );

    return replacedSingleBabbling ? acc : acc + 1;
  }, 0);
}

```